Kuroba/app/src/main/java/com/github/adamantcheese/chan/features/embedding/embedders/DlsiteEmbedder.java

```diff
@@ -0,0 +1,106 @@
+package com.github.adamantcheese.chan.features.embedding.embedders;
+
+import android.graphics.Bitmap;
+import android.util.JsonReader;
+
+import com.github.adamantcheese.chan.BuildConfig;
+import com.github.adamantcheese.chan.core.model.PostImage;
+import com.github.adamantcheese.chan.core.net.NetUtilsClasses;
+import com.github.adamantcheese.chan.core.repository.BitmapRepository;
+import com.github.adamantcheese.chan.features.embedding.EmbedResult;
+import com.github.adamantcheese.chan.utils.StringUtils;
+import com.google.common.io.Files;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import okhttp3.HttpUrl;
+
+public class DlsiteEmbedder
+        extends JsonEmbedder {
+
+    private final Pattern DLSITE_PATTERN =
+            Pattern.compile("(?:http|www|dlsite)[^>\\s]+[rR][jJ](\\d{6})(?:\\.html)?");
```

Comment:
Did you intend for something like `http=RJ123456` match? You should probably change this regex to be more restrictive and match the entire link. Maybe something like `(?:https?://www\.dlsite).+[rR][jJ](\d{6})(?:\.html)?` would work better?

Reply:
Fixed by making the dlsite.com portion of the regex `https?://(?:www\.)?dlsite\.com.+?[rR][jJ](\d{6})(?:\.html)?`. Now it shouldn't match `http=RJ123456`, although the `RJ123456` portion would still be matched by the rjcode regex.